src/land/madis_snow2ioda.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+#
+# (C) Copyright 2021-2024 EMC/NCEP/NWS/NOAA
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+import os
+import argparse
+import netCDF4 as nc
+import numpy as np
+from datetime import datetime
+
+import pyiodaconv.ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from pyiodaconv.orddicts import DefaultOrderedDict
+
+os.environ["TZ"] = "UTC"
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("stationElevation", "float"),
+    ("dateTime", "long")
+]
+
+obsvars = {
+    'snow_depth': 'totalSnowDepth',
+}
+
+AttrData = {
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'totalSnowDepth': ['Location'],
+}
+
+iso8601_string = 'seconds since 1970-01-01T00:00:00Z'
+epoch = datetime.fromisoformat(iso8601_string[14:-1])
```

Comment:
just getting to this.

I'm going to change this so it will inherit these from a central location, will make it easier if these ever change (the reference epoch)